go/vt/vtgate/buffer/buffer_test.go

```diff
@@ -782,3 +795,66 @@ func testShutdown1(t *testing.T, fail failover) {
 		t.Fatal(err)
 	}
 }
+
+func TestParallelRangeIndex(t *testing.T) {
+	suite := []struct {
+		max         int
+		concurrency int
+		calls       []int
+	}{
+		{
+			max:         0,
+			concurrency: 0,
+			calls:       []int{},
+		},
+		{
+			max:         100,
+			concurrency: 0,
+			calls:       []int{},
+		},
+		{
+			max:         9,
+			concurrency: 3,
+			calls:       []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		},
+		{
+			max:         0,
+			concurrency: 10,
+			calls:       []int{0},
+		},
+		{
+			max:         9,
+			concurrency: 9,
+			calls:       []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		},
+	}
+
+	for idx, tc := range suite {
+		name := fmt.Sprintf("%d_max%d_concurrency%d", idx, tc.max, tc.concurrency)
+		t.Run(name, func(t *testing.T) {
```

Comment:
small issue around variable shadowing here (for `tc`): https://stackoverflow.com/a/33459174

Reply:
@ajm188 not sure I follow. the tests are working properly. the t.Run call and goroutines below terminate before the next loop iteration.